scipy/interpolate/tests/test_bary_rational.py

```diff
@@ -235,3 +239,62 @@ def test_spiral_cleanup(self):
         assert np.sum(np.abs(r.residues()) < 1e-14) < n_spurious
         # check accuracy
         assert_allclose(r(z), np.tan(np.pi*z/2), atol=6e-12, rtol=3e-12)
+
+
+class TestFloaterHormann:
+    def runge(self, z):
+        return 1/(1 + z**2)
+
+    def scale(self, n, d):
+        return (-1)**(np.arange(n) + d) * factorial(d)
+
+    def test_iv(self):
+        with pytest.raises(ValueError, match="`d`"):
+            FloaterHormann([0], [0], d=-1)
+        with pytest.raises(ValueError, match="`d`"):
+            FloaterHormann([0], [0], d=10)
+
+    @pytest.mark.parametrize("d,expected", [
+        (0, [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
+        (1, [1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1]),
+        (2, [1, 3, 4, 4, 4, 4, 4, 4, 4, 3, 1]),
+        (3, [1, 4, 7, 8, 8, 8, 8, 8, 7, 4, 1]),
+        (4, [1, 5, 11, 15, 16, 16, 16, 15, 11, 5, 1])
+    ])
+    def test_uniform_grid(self, d, expected):
+        # Check against explicit results on an uniform grid
+        x = np.arange(11)
+        r = FloaterHormann(x, 0.0*x, d=d)
+        assert_allclose(r.weights*self.scale(x.size, d), expected,
+                        rtol=1e-15, atol=1e-15)
+
+    @pytest.mark.parametrize("d", range(10))
+    def test_runge(self, d):
+        x = np.linspace(0, 1, 51)
+        xx = np.linspace(0, 1, 1001)
+        y = self.runge(x)
+        h = x[1] - x[0]
+
+        r = FloaterHormann(x, y, d=d)
+
+        tol = 10*h**(d+1)
+        assert_allclose(r(xx), self.runge(xx), atol=1e-10, rtol=tol)
+        # check interpolation property
+        assert_equal(r(x), self.runge(x))
```

Comment:
nice :-). Makes me want to try the "bad" Runge function, `1 / (1 + 25 * x**2)` on `x\in [-1, 1]`.